plugins/snowflake/dbt/include/snowflake/macros/catalog.sql

```diff
@@ -2,59 +2,66 @@
+                    'The key used to cluster this table' as "stats:clustering_key:description",
+                    (clustering_key is not null) as "stats:clustering_key:include",
 
-            'Row Count' as "stats:row_count:label",
-            row_count as "stats:row_count:value",
-            'An approximate count of rows in this table' as "stats:row_count:description",
-            (row_count is not null) as "stats:row_count:include",
+                    'Row Count' as "stats:row_count:label",
+                    row_count as "stats:row_count:value",
+                    'An approximate count of rows in this table' as "stats:row_count:description",
+                    (row_count is not null) as "stats:row_count:include",
 
-            'Approximate Size' as "stats:bytes:label",
-            bytes as "stats:bytes:value",
-            'Approximate size of the table as reported by Snowflake' as "stats:bytes:description",
-            (bytes is not null) as "stats:bytes:include"
+                    'Approximate Size' as "stats:bytes:label",
+                    bytes as "stats:bytes:value",
+                    'Approximate size of the table as reported by Snowflake' as "stats:bytes:description",
+                    (bytes is not null) as "stats:bytes:include"
 
-        from information_schema.tables
+                from {{ adapter.quote_as_configured(database, "database") }}.information_schema.tables
 
-    ),
+            ),
 
-    columns as (
+            columns as (
 
-        select
+                select
+                    table_catalog as "table_database",
+                    table_schema as "table_schema",
+                    table_name as "table_name",
+                    null as "table_comment",
 
-            table_schema as "table_schema",
-            table_name as "table_name",
-            null as "table_comment",
+                    column_name as "column_name",
+                    ordinal_position as "column_index",
+                    data_type as "column_type",
+                    null as "column_comment"
 
-            column_name as "column_name",
-            ordinal_position as "column_index",
-            data_type as "column_type",
-            null as "column_comment"
+                from {{ adapter.quote_as_configured(database, "database") }}.information_schema.columns
 
-        from information_schema.columns
+            )
 
-    )
-
-    select *
-    from tables
-    join columns using ("table_schema", "table_name")
-    where "table_schema" != 'INFORMATION_SCHEMA'
-    order by "column_index"
+            select *
+            from tables
+            join columns using ("table_database", "table_schema", "table_name")
+            where "table_schema" != 'INFORMATION_SCHEMA'
+              and "table_database" = {{ adapter.quote_as_configured(database, "database").replace('"', "'") }}
+            order by "column_index"
+        )
+        {% if not loop.last %} union all {% endif %}
```

Comment:
There's an outstanding limitation of Snowflake that causes some queries against information_schema.tables to fail (via https://github.com/fishtown-analytics/dbt/issues/1127). Let's leave this as-is (I think it's slick) but let's be mindful of this change w/r/t that issue going forwards.

Reply:
Given the explanation in that issue of the reason behind the behavior, it sounds like this will be no change really, right? This query basically just adds a where-clause that would be ignored, and prefixes `information_schema` with the database name.